normandy/recipes/models.py

```diff
@@ -121,16 +150,40 @@ def __repr__(self):
     def __str__(self):
         return self.name
 
-    def save(self, *args, **kwargs):
+    def save(self, *args, skip_last_updated=False, **kwargs):
         # Increment the revision ID and save the reversion if we've
         # changed
         if self.is_dirty(check_relationship=True):
+            dirty_fields = self.get_dirty_fields(check_relationship=True)
+
+            # Any change will invalid the signature
```

Comment:
Doing this last won't help, unfortunately. But I think we can bail early to make sure things work out. I'll write some tests to be sure.